var/spack/repos/builtin/packages/elemental/package.py

```diff
@@ -0,0 +1,112 @@
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class Elemental(CMakePackage):
+    """Elemental: Distributed-memory dense and sparse-direct linear algebra and optimization library."""
+
+    homepage = "http://libelemental.org"
+    url      = "https://github.com/elemental/Elemental/archive/v0.87.6.tar.gz"
+
+    version('0.87.6', '9fd29783d45b0a0e27c0df85f548abe9')
+
+    variant('debug', default=False, 
+            description='Builds a debug version of the libraries')
+    variant('shared', default=True, 
+            description='Enables the build of shared libraries')
+    variant('hybrid', default=True, 
+            description='Elemental: make use of OpenMP within MPI packing/unpacking')
+    variant('c_interface', default=False, 
+            description='Elemental: build C interface')
+    variant('python_package', default=False, 
+            description='Elemental: install Python interface')
+    variant('disable_parmetis', default=True, 
+            description='Elemental: disable ParMETIS')
+    variant('disable_quad', default=True, 
+            description='Elemental: disable quad precision')
+    variant('int64', default=False, 
+            description='Elemental: use 64bit integers')
+    variant('int64_blas', default=False, 
+            description='Elemental: use 64bit integers for BLAS')
+    variant('scalapack', default=False,
+            description='Elemental: build with ScaLAPACK library')
+
+    depends_on('cmake', type='build')
+    depends_on('openblas +openmp', when='~int64_blas')
+    depends_on('metis')
+    depends_on('metis +int64', when='+int64')
+    depends_on('mpi')
+    depends_on('netlib-scalapack', when='+scalapack ~int64_blas')
+
+    def build_type(self):
+        """Returns the correct value for the ``CMAKE_BUILD_TYPE`` variable
+        :return: value for ``CMAKE_BUILD_TYPE``
+        """
+        if '+debug' in self.spec:
+            return 'Debug'
+        else:
+            return 'Release'
+
+    def cmake_args(self):
+        args = ['-DCMAKE_INSTALL_MESSAGE:STRING=LAZY',
+                '-DEL_PREFER_OPENBLAS:BOOL=TRUE',
+                '-DEL_DISABLE_SCALAPACK:BOOL={0}'.format((
+                    'ON' if '+scalapack' in self.spec else 'OFF')),
+                '-DGFORTRAN_LIB=libgfortran.so',
```

Comment:
Why is this needed?  Normally, you do not have to pass the location or name of `libgfortran.so` to builds.  This looks like a wonky thing in this build; but it will ultimately bite us when someone tries to build with non-GCC.  I'd like to get more explanation of why it is here, then maybe we can find a better way.